src/main/java/de/tum/in/www1/artemis/service/exam/ExamDeletionService.java

```diff
@@ -240,4 +256,33 @@ public void deleteTestRun(Long testRunId) {
         log.info("Request to delete Test Run {}", testRunId);
         studentExamRepository.deleteById(testRunId);
     }
+
+    /**
+     * Get the exam deletion summary for the given exam.
+     *
+     * @param examId the ID of the exam for which the deletion summary should be fetched
+     * @return the exam deletion summary
+     */
+    public ExamDeletionSummaryDTO getExamDeletionSummary(@NotNull long examId) {
+        Exam exam = examRepository.findOneWithEagerExercisesGroupsAndStudentExams(examId);
+        long numberOfBuilds = exam.getExerciseGroups().stream().flatMap(group -> group.getExercises().stream())
+                .filter(exercise -> ExerciseType.PROGRAMMING.equals(exercise.getExerciseType()))
+                .mapToLong(exercise -> buildJobRepository.countBuildJobsByExerciseIds(List.of(exercise.getId()))).sum();
+
+        Channel channel = channelRepository.findChannelByExamId(examId);
+        Long conversationId = channel.getId();
+
+        List<Long> postIds = postRepository.findAllByConversationId(conversationId).stream().map(Post::getId).toList();
+        long numberOfCommunicationPosts = postIds.size();
+        long numberOfAnswerPosts = answerPostRepository.countAnswerPostsByPostIdIn(postIds);
+
+        Set<StudentExam> studentExams = exam.getStudentExams();
+        long numberRegisteredStudents = studentExams.size();
+
+        long notStartedExams = studentExams.stream().filter(studentExam -> !studentExam.isStarted()).count();
+        long startedExams = studentExams.stream().filter(studentExam -> studentExam.isStarted() && !studentExam.isSubmitted()).count();
+        long submittedExams = studentExams.stream().filter(StudentExam::isSubmitted).count();
```

Comment:
Sadly `started` and `submitted` have type `Boolean` instead of `boolean` and they can be null (see other usages of the getters)